stable/database/files/nuobackup

```diff
@@ -64,19 +71,32 @@ function execute_backup() {
    retval=$?
    if [ $retval != 0 ]; then
       echo >&2 "Error running hotcopy $retval"
+      # Store failed backupset in KV store
+      if [ "$type" = "full" ] && [ -n "$backup_group" ]; then
+         backupset=$(get_current_backup)
+         if [ -n "$backupset" ]; then
+            failed=$(nuocmd get value --key "$NUODB_BACKUP_KEY/$db_name/${backup_group}/failed")
+            nuocmd set value \
+               --key "$NUODB_BACKUP_KEY/$db_name/${backup_group}/failed" \
+               --value "${failed} ${backupset}" --expected-value "${failed}"
+         fi
+      fi 
```

Comment:
Could the reason for backup failure also cause the key value command to fail.  I guess then you are no worse a position than without these changes.

Reply:
Correct, for this to work, we need a cooperative admin layer.